src/utils/thread_helper.cpp

```diff
@@ -224,6 +225,26 @@ void CopyMdnsResponseCounters(const MdnsResponseCounters &from, threadnetwork::T
     to->set_aborted_count(from.mAborted);
     to->set_invalid_state_count(from.mInvalidState);
 }
+
+threadnetwork::TelemetryData_EpskcState GetEpskcState(otInstance *aInstance)
+{
+    threadnetwork::TelemetryData::EpskcState epskcState;
+
+    VerifyOrExit(otBorderAgentIsEphemeralKeyActive(aInstance),
+                 epskcState = threadnetwork::TelemetryData::EPSKC_STATE_INACTIVE);
```

Comment:
I'm wondering if we could represent the two states as-is instead of merging them into one state? That could be more straightforward and less data loss.

Reply:
Updated the proto to send both epskc active state and border agent state.